beacon-chain/rpc/eth/beacon/handlers_validator.go

```diff
@@ -324,6 +326,152 @@ func (s *Server) GetValidatorBalances(w http.ResponseWriter, r *http.Request) {
 	httputil.WriteJson(w, resp)
 }
 
+// GetValidatorIdentities returns a filterable list of validators identities.
+func (s *Server) GetValidatorIdentities(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetValidatorIdentities")
+	defer span.End()
+
+	stateId := r.PathValue("state_id")
+	if stateId == "" {
+		httputil.HandleError(w, "state_id is required in URL params", http.StatusBadRequest)
+		return
+	}
+	st, err := s.Stater.State(ctx, []byte(stateId))
+	if err != nil {
+		shared.WriteStateFetchError(w, err)
+		return
+	}
+
+	var rawIds []string
+	err = json.NewDecoder(r.Body).Decode(&rawIds)
+	switch {
+	case errors.Is(err, io.EOF):
+		httputil.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	case err != nil:
+		httputil.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	ids, ok := decodeIds(w, st, rawIds, true /* ignore unknown */)
+	if !ok {
+		return
```

Comment:
shouldn't we be returning some http error here?

Reply:
It's already set inside `decodeIds`. I agree it's very non-intuitive, maybe we should have a rule that we return `RpcError` from every non-top level function and the top-level handler is responsible for actually setting http values.

Reply:
yeah lets try to add it at the top level

Reply:
I can do that in a separate PR for all Beacon API handlers if that's OK with you